### PR TITLE
update cloud-provider-vsphere

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/area control-plane
/kind technical-debt
/priority normal
/platform vsphere

**What this PR does / why we need it**:
The [cloud-provider-vsphere](https://github.com/kubernetes/cloud-provider-vsphere) is switching from INI style config file to YAML. The `config-provider-config` is now created as YAML file.
Additionally the latest bugfixes from release v1.2.1 are included in the new image of cloud-provider-vsphere.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
updated dependency cloud-provider-vsphere to v1.2.1
```

### CODE REVIEW

File: charts/internal/cloud-provider-config/templates/cloud-provider-config.yaml [hunk 1]
```diff
@@ -5,63 +5,75 @@ metadata:
...
+    global:
+      soapRoundtripCount: {{ .Values.soapRoundtripCount }}
+      ipFamily:
+      - ipv4
+      {{- if .Values.caFile }}
+      caFile: "{{ .Values.caFile }}"
+      {{- end }}
+      {{- if .Values.caFile }}
+      thumbprint: "{{ .Values.thumbprint }}"
+      {{- end }}
```
Comment (mandelsoft) [inline charts/internal/cloud-provider-config/templates/cloud-provider-config.yaml:17]: The same condition?

File: pkg/controller/controlplane/valuesprovider.go [hunk 2]
```diff
@@ -455,7 +455,7 @@ func (vp *valuesProvider) getConfigChartValues(
 		"serverName":   serverName,
 		"serverPort":   port,
 		"insecureFlag": region.VsphereInsecureSSL,
-		"datacenters":  strings.Join(helper.CollectDatacenters(region), ","),
+		"datacenters":  helper.CollectDatacenters(region),
```
Comment (mandelsoft) [inline pkg/controller/controlplane/valuesprovider.go:458]: Why are there helper packages called `helper` and `helpers` i the same project.

Comment (MartinWeindel) [reply pkg/controller/controlplane/valuesprovider.go:458]: Nice hint. Will note it for a refactoring task, but not in this PR.

File: charts/internal/cloud-provider-config/templates/cloud-provider-config.yaml (hunk 1 above)
Comment (MartinWeindel) [reply charts/internal/cloud-provider-config/templates/cloud-provider-config.yaml]: This was a copy/paste bug

**Review by mandelsoft — APPROVED** [review]

**Pull request merged**